File: all/tessoku-book/tessoku_book_ay/tessoku_book_ay.py
q = int(input())

stack = []
for _ in range(q):
  t = input().split()
  if len(t) == 2: x = t[1]
  t = int(t[0])

  if t == 1:
    stack.append(x)
  
  if t == 2:
    print(stack[-1])
  
  if t == 3:
    stack.pop()

File: all/tessoku-book/tessoku_book_ay/tessoku_book_ay.cs
using static System.Console;

public class A
{
  static void Main()
  {
    int q = int.Parse(ReadLine());

    var st = new Stack<string>();
    for(int _ = 0; _ < q; _++)
    {
      string[] t = ReadLine().Split();
      if(t.Length == 2) {
        st.Push(t[1]);
      } else {
        if(t[0] == "2") {
          WriteLine(st.Peek());
        } else if(t[0] == "3") {
          st.Pop();
        }
      }
    }
  }
}
